tests/integration/Owncloud/OcisPhpSdk/OcisResourceTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace integration\Owncloud\OcisPhpSdk;
+
+require_once __DIR__ . '/OcisPhpSdkTestCase.php';
+
+use Owncloud\OcisPhpSdk\Drive; // @phan-suppress-current-line PhanUnreferencedUseNormal it's used in a comment
+use Owncloud\OcisPhpSdk\DriveOrder;
+use Owncloud\OcisPhpSdk\DriveType;
+use Owncloud\OcisPhpSdk\Ocis;
+use Owncloud\OcisPhpSdk\OcisResource; // @phan-suppress-current-line PhanUnreferencedUseNormal it's used in a comment
+use Owncloud\OcisPhpSdk\OrderDirection;
+
+class OcisResourceTest extends OcisPhpSdkTestCase
+{
+    public function testGetResources(): void
+    {
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis($this->ocisUrl, $token, ['verify' => false]);
+        /**
+         * @var Drive $personalDrive
+         */
+        $personalDrive = $ocis->getMyDrives(
+            DriveOrder::NAME,
+            OrderDirection::ASC,
+            DriveType::PERSONAL
+        )[0];
+        $personalDrive->uploadFile('somefile.txt', 'some content');
+        $personalDrive->uploadFile('secondfile.txt', 'some other content');
+        $personalDrive->createFolder('subfolder');
+        $this->createdResources[] = '/somefile.txt';
+        $this->createdResources[] = '/secondfile.txt';
+        $this->createdResources[] = '/subfolder';
+
+        $resources = $personalDrive->getResources();
+        $this->assertCount(3, $resources);
+
+        /**
+         * @var OcisResource $resource
+         */
+        foreach ($resources as $resource) {
+            $this->assertMatchesRegularExpression(
+                "/^" . $this->getFileIdRegex() . "$/i",
+                $resource->getId()
+            );
+            $this->assertMatchesRegularExpression(
+                "/^\"[a-f0-9:\.]{1,32}\"$/",
+                $resource->getEtag()
+            );
+            $this->assertMatchesRegularExpression(
+                "?^" . $this->ocisUrl . '/f/'. $this->getFileIdRegex() . "$?i",
+                $resource->getPrivatelink()
+            );
+            try {
+                $this->assertEquals('file', $resource->getType());
+            } catch (\Exception $e) {
+                $this->assertEquals('folder', $resource->getType());
+            }
```

Comment:
phpunit has `logicalOr` - maybe this can be something like:
```
$this-logicalOr($this->assertEquals('file', $resource->getType()), $this->assertEquals('folder', $resource->getType()));
```
and similar for some other try-catch here.

Reply:
done